src/main/java/sorald/sonar/RuleVerifier.java

```diff
@@ -172,8 +211,48 @@ public void reportIssue(AnalyzerMessage analyzerMessage) {
             messages.add(analyzerMessage);
         }
 
+        @Override
+        public void setSensorContext(SensorContext context) {
+            this.context = context;
+            super.setSensorContext(context);
+        }
+
+        public SensorContext getContext() {
+            return context;
+        }
+
         public List<AnalyzerMessage> getMessages() {
-            return Collections.unmodifiableList(messages);
+            return messages.stream()
+                    .filter(message -> postFilter.accept(getRuleKey(message), message))
+                    .collect(Collectors.toList());
+        }
+
+        public JavaClasspath getClasspath() {
+            return cp;
+        }
+
+        private static RuleKey getRuleKey(AnalyzerMessage message) {
+            return RuleKey.of("java", "S" + Checks.getRuleKey(message.getCheck().getClass()));
```

Comment:
Another sign that someday we should change our ruleKeys from String to Integer.

Reply:
Indeed, I really think we should get on that sooner rather than later. It's getting rather annoying.

Reply:
> from String to Integer

* from Integer to String :D

Reply:
I've turned this into a helper function `Checks.toSonarRuleKey` and added a TODO that we should remove it by simply using the Sonar keys in Sorald.